### PR TITLE
127-view-all-products

### LINKED ISSUE #127: View for All Products
### Feature Description

When a user is on the home page and selects `View All Products` there is no view for them to land on.
### Recommended Actions

> Refer to the [project requirements](https://github.com/TeamCharles/bangazon-inc/blob/master/projects/INITIAL_BANGAZON_SITE.md#requirements) before filling out this section.
1. Create a View and Controller for Products/Index
2. Copy the razor template from ProductTypes/List
3. Use the ProductList ViewModel to build the view from the controller's Index method

### PR DESCRIPTION
**Pull request by MattKraatz**
## Status

**READY**
## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.
## Description

Provide a view of all products when the user selects "View All Products"
## Resolves Issue Number

Resolves #127 
## Deploy Notes

``` sh
git pull --prune
git checkout 127-view-all-products
dotnet run
```
## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
1. Navigate to the home page
2. Select "View All Products"
## Impacted Files in Application

List general components of the application that this PR will affect:
- Controllers/ProductsController
- Views/Products/Index

### CODE REVIEW

**Comment by garrett-vangilder** [comment]
LGTM

**Comment by daynewright** [comment]
LGTM.   Added table formatting for the view.

File: Views/ProductTypes/Buy.cshtml [hunk 1]
```diff
@@ -3,12 +3,21 @@
...
-  </div>
-}
\ No newline at end of file
+  <tr>
+      <td><a asp-action="List" asp-route-id="@type.ProductTypeId">@Html.DisplayFor(model => type.Label)</a></td>
+      <td>(@Html.DisplayFor(model => type.Quantity))</td>
+  </tr>
+}
+    </tbody>
+</table>
```
Comment (garrett-vangilder) [inline Views/ProductTypes/Buy.cshtml:23]: Apparently old school tech messes up without an empty line character. http://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-file

But everything LGTM

**Comment by garrett-vangilder** [comment]
LGTM

**Review by Anulfo — COMMENTED** [review]
LGTM. Pretty Table!!

**Review by garrett-vangilder — APPROVED** [review]

**Pull request merged**